### ISSUE TITLE
After update to MIUI 12,Text is cut off

### ISSUE THREAD

**Issue by jinshiyi11** [facebook/react-native#29259]
## Description
After update to MIUI 12, Text is cut off.
![image](https://user-images.githubusercontent.com/3433961/86214160-5d766d00-bbad-11ea-9668-b11786bc8825.png)
the actual text is QQ,but only Q is displayed

## React Native version:
React Native version: 0.60.5
model： MI 10 Pro
android：10
MIUI：12.0.1
Fingerprint -> Xiaomi/cmi/cmi:10/QKQ1.191117.002/V12.0.1.0.QJACNXM:user/release-keys

## Steps To Reproduce
updrade XiaoMi Phone's system to MIUI 12

## Expected Results
full text is expected to display,like below  
![image](https://user-images.githubusercontent.com/3433961/86214415-bfcf6d80-bbad-11ea-9e38-f1ed600d1840.png)

**Comment by xiaohui-gsv** [comment]
yeah, i have the same issue in miui 12 but it is fine in miui 10 and 9
![image](https://user-images.githubusercontent.com/38554198/86219157-6159bd80-bbb4-11ea-9d6c-d754c7db1c00.png)
.
The phone number and the word 'message' disappeared. Not sure it's a issue in react-native or miui.

**Comment by jinshiyi11** [comment]
currently ,we hook Text render to make it work correctly.
`
const originTextRender = Text.render;  

Text.render = function render(props, ref) {  

        return originTextRender.apply(this, [{ ...props,  style: [isMiUi12() ? {fontFamily:''} : null, props.style] }, ref]);  

    };  
`

**Comment by sapjax** [comment]
This is caused by the default FontFamily of MIUI 12: "小米兰亭 PRO".

**Comment by sapjax** [comment]
@jinshiyi11 
What is your code to detect MIUI12 in `isMiUi12`?

**Comment by bashen1** [comment]
The same issue in miui 12

**Comment by Foveluy** [comment]
> currently ,we hook Text render to make it work correctly.
> `
> const originTextRender = Text.render;
> 
> Text.render = function render(props, ref) {
> 
> ```
>     return originTextRender.apply(this, [{ ...props,  style: [isMiUi12() ? {fontFamily:''} : null, props.style] }, ref]);  
> 
> };  
> ```
> 
> `

cannot fix the problem

**Comment by AKB48** [comment]
It did work to set a default property - {fontFamily: ''} in style property of Text

**Comment by KingAmo** [comment]
same here,

**Comment by everfire130** [comment]
The same issue in RN0.63.2

**Comment by jinshiyi11** [comment]
> @jinshiyi11
> What is your code to detect MIUI12 in `isMiUi12`?

the code likes below

`
import {  Platform, NativeModules, TextInput, Text } from 'react-native';

function isMiUi12() {
    if(Platform.OS === 'android' && NativeModules.PlatformConstants){
        const fingerprint = NativeModules.PlatformConstants.Fingerprint && NativeModules.PlatformConstants.Fingerprint.toLowerCase();
        console.log(`fingerprint:${fingerprint}`);
        // MiUi12 fingerprint:Xiaomi/cmi/cmi:10/QKQ1.191117.002/V12.0.1.0.QJACNXM:user/release-keys
        if(fingerprint && fingerprint.startsWith('xiaomi') && fingerprint.indexOf('v12')!==-1){
            return true;
        }
    }
   
    return false;
}
`

**Comment by fangasvsass** [comment]
any update ?

**Comment by eagleonefight** [comment]
same here

**Comment by monomichael** [comment]
This looks more like a MIUI 12 issue to me.

We had the same problem, but when we changed the system's font weight to another value (in Settings - Display - Text size & font weight), the issue disappeared. When we changed it back to the original value, the rendering remained perfect.

I'm reaching out to the MIUI team and I'll post an update if I receive any response.

**Comment by inkCrazy** [comment]
最后你们怎么解决问题的，RN0.62.2  小米10， MIUI12.0.8同样问题。

**Comment by magicismight** [comment]
```
if (Platform.OS === 'android' && NativeModules.PlatformConstants) {
  const fingerprint = NativeModules.PlatformConstants.Fingerprint;

  if (fingerprint?.match(/^(xiaomi|redmi|mi|mix|poco).*\/v12\..*/i)) {
    const originTextRender = Text.render;

    Text.render = function render(props, ref) {
      return originTextRender.apply(this, [
        {
          ...props,
          style: [{ fontFamily: '' }, props.style]
        },
        ref
      ]);
    };
  }
}
```

**Comment by inkCrazy** [comment]
> ```
> if (Platform.OS === 'android' && NativeModules.PlatformConstants) {
>   const fingerprint = NativeModules.PlatformConstants.Fingerprint;
> 
>   if (fingerprint?.match(/^(xiaomi|redmi|mi|mix|poco).*\/v12\..*/i)) {
>     const originTextRender = Text.render;
> 
>     Text.render = function render(props, ref) {
>       return originTextRender.apply(this, [
>         {
>           ...props,
>           style: [{ fontFamily: '' }, props.style]
>         },
>         ref
>       ]);
>     };
>   }
> }
> ```

这个是写在app.js里面，全局设置字体？

**Comment by hussainahmad** [comment]
> ```
> if (Platform.OS === 'android' && NativeModules.PlatformConstants) {
>   const fingerprint = NativeModules.PlatformConstants.Fingerprint;
> 
>   if (fingerprint?.match(/^(xiaomi|redmi|mi|mix|poco).*\/v12\..*/i)) {
>     const originTextRender = Text.render;
> 
>     Text.render = function render(props, ref) {
>       return originTextRender.apply(this, [
>         {
>           ...props,
>           style: [{ fontFamily: '' }, props.style]
>         },
>         ref
>       ]);
>     };
>   }
> }
> ```

@magicismight where this code be put?

**Comment by yorickshan** [comment]
```fontFamily: 'System'```

it works for me

**Comment by zhuminjunivan** [comment]
> This looks more like a MIUI 12 issue to me.
> 
> We had the same problem, but when we changed the system's font weight to another value (in Settings - Display - Text size & font weight), the issue disappeared. When we changed it back to the original value, the rendering remained perfect.
> 

我也是这个问题，但是我清除app数据就好了。这就让我特别奇怪。一开始我们是有客户用华为手机然后反馈的，然后小米也复现了。oppo和vivo没有问题，仔细对比了一下，感觉小米的默认字体加粗比oppo的加粗粗好多。不知道怎么解决好。我现在一脸懵。
